src/picongpu/include/particles/ionization/ionizationMethods.hpp

```diff
@@ -63,7 +63,7 @@ namespace ionization
 
             /* each thread sets the multiMask hard on "particle" (=1) */
             childElectron[multiMask_] = 1;
-            const uint32_t weighting = parentIon[weighting_];
```

Comment:
No because we allow implicit cast, I am searching for a flag to disallow implicit floating point to int cast.
Disallow all implicit cast is currently to much :-(